client/components/theme/index.jsx

```diff
@@ -61,11 +61,44 @@ const Theme = React.createClass( {
 		actionLabel: React.PropTypes.string
 	},
 
+	areButtonContentsEqual( nextButtonContents, buttonContents ) {
+		if ( ! isEqual( Object.keys( nextButtonContents ), Object.keys( buttonContents ) ) ) {
+			return false;
+		}
+
+		return every( buttonContents, ( option, name ) => {
```

Comment:
Good question. Well `getUrl` implicitly depends on the `selectedSiteId`, so there is a chance that if we remove the `every` check, theme options will persist when switching between sites (meaning that e.g. the newly selected site's active theme is displayed with an inactive theme's options, coming from that same theme on the previously selected site).

However, it seems there are a couple of other ancestral components that get re-rendered on switching sites, so this doesn't currently seem to be an issue. I think I'm going to remove it.